### PR TITLE
User signup

### PR DESCRIPTION
**Pull request by FayeCarter**
Sign up feature completed, please check it out?

### CODE REVIEW

File: bnb.rb [hunk 1]
```diff
@@ -22,4 +25,15 @@ class BnB < Sinatra::Base
     result = Space.create(name: params['name'])
     "#{result.name} has been listed"
   end
+
+  get '/users/new' do
+    erb :'users/new'
+  end
+
+  post '/users' do    
+    user = User.create(first_name: params['firstname'], last_name: params['lastname'], password: params['password'], email: params['email'])
```
Comment (PhilipVigus) [inline bnb.rb:34]: Would it be worth formatting this to reduce the line length? (each argument on its own line to improve readability)

File: lib/user.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @first_name = first_name
+    @last_name = last_name
+    @email = email
+    @password = password
+  end
+
+  def self.create(first_name:, last_name:, email:, password:)
+    hashed_password = BCrypt::Password.create(password)
+    connection = PG.connect(dbname: 'bnb_test')
+    result = connection.exec("INSERT INTO users (firstname, lastname, email, passworddigest) VALUES('#{first_name}','#{last_name}','#{email}','#{hashed_password}') RETURNING id, firstname, lastname, email, passworddigest;")
```
Comment (PhilipVigus) [inline lib/user.rb:19]: Should multi-word columns be underscore separated or not? We've been using underscores. Don't think it matters which we do as long as we're consistent.

File: lib/user.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
```
Comment (PhilipVigus) [inline lib/user.rb:1]: Really good code here, nice and clear and well thought out.

File: spec/features/usersignup_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
```
Comment (PhilipVigus) [inline spec/features/usersignup_spec.rb:1]: Really good feature test 😀

File: bnb.rb (hunk 1 above)
Comment (FayeCarter) [reply bnb.rb:34]: Sounds good, we will change this

File: spec/set_up_test_database.rb [hunk 5]
```diff
@@ -8,4 +8,5 @@ def setup_test_database
 
   # Clear the tables
   connection.exec("TRUNCATE spaces CASCADE;")
+  connection.exec("TRUNCATE users CASCADE;")
```
Comment (PhilipVigus) [inline spec/set_up_test_database.rb:11]: Probably worth considering combining these lines into "TRUNCATE spaces, users CASCADE"

File: spec/features/usersignup_spec.rb (hunk 4 above)
Comment (FayeCarter) [reply spec/features/usersignup_spec.rb:1]: Thanks! we pul a lot of thought into it 🙂

File: spec/set_up_test_database.rb (hunk 5 above)
Comment (FayeCarter) [reply spec/set_up_test_database.rb:11]: Sure, no problem

File: lib/user.rb (hunk 2 above)
Comment (hturnbull93) [reply lib/user.rb:19]: I thought the general convention for multi-word columns in sql tables was just to ram them all together :shrug:

File: views/index.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
```
Comment (PhilipVigus) [inline views/index.erb:1]: Perfection 🥰

**Comment by PhilipVigus** [comment]
Generally really, really good. There are a couple of bits and pieces around formatting overly long lines to make them more readable, and one consistency thing regarding database column naming, but that's it 👍

**Pull request merged**